raft/log_test.go

```diff
@@ -281,9 +281,7 @@ func TestLog_WriteEntriesTo_Snapshot(t *testing.T) {
 	}()
 
 	// Wait for buffer to be written to.
-	for buf.Len() == 0 {
-		runtime.Gosched()
-	}
+	time.Sleep(1 * time.Second)
```

Comment:
I noticed intermittently incorrect results. For instance, I just noticed another with the same pattern in the same file:
```
--- FAIL: TestLog_WriteEntriesTo-8 (0.02s)
	log_test.go:267: unexpected entry count: 1
```
And I'm not sure I understand what you mean by "Len() has been locked". But what seems to be happening is that not all of the data is being written to 'buf' before it gets processed.